File: src/main/java/com/hazza/leetcode/SegmentThree/No104MaximumDepthOfBinaryTree.java
package com.hazza.leetcode.SegmentThree;

import com.hazza.leetcode.datastructure.TreeNode;

/**
 * Created with IntelliJ IDEA.
 * Description: https://leetcode.com/problems/maximum-depth-of-binary-tree/description/
 * User: HazzaCheng
 * Contact: hazzacheng@gmail.com
 * Date: 18-1-16
 * Time: 3:16 PM
 */
public class No104MaximumDepthOfBinaryTree {
    public int maxDepth(TreeNode root) {
        if (root == null) return 0;
        return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }
}

File: src/main/python/leetcode_by_python/tree/No104_Maximum_Depth_of_Binary_Tree.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019-09-11
# https://leetcode.com/problems/maximum-depth-of-binary-tree/
from src.main.python.leetcode_by_python.data_structure.TreeNode import TreeNode


class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if not root:
            return 0

        return max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1

